### PR TITLE
Closes #413

### LINKED ISSUE #413: Implement easy cobra commands, fix command outputs
At the moment, the source utilises Cobra CLI in a simple form when it comes to adding the commands to it. In the main file for the `sztp-agent` you have the following:

https://github.com/opiproject/sztp/blob/33b5b5ce949c76c17786e71e9dc933443b294e15/sztp-agent/main.go#L28C1-L47C2

The individual use of `c.AddCommand(...)` can be quite cumbersome in the future where the CLI grows. So I'd like to propose a solution for this:

1. Create a global `var Commands []*cobra.Command` slice
2. Have all the cobra commands setup inside an `init()` func instead of `NewXCommand`
3. All the `init()` funcs for all commands append themselves to the commands slice
4. Loop through the slice in main:
```go
for _, cmd := range Commands {
   c.AddCommand(cmd)
}
```
Additionally, there seems to be something I think could be changed. Every time a command might fail, it runs prints the help text. I was testing it with my new command I created for generating `systemd` unit files:
```
~/GolandProjects/sztp/sztp-agent git:[main]
./sztp-agent systemd
Error: creating unit file /etc/systemd/system/sztp-agent.service: open /etc/systemd/system/sztp-agent.service: permission denied
Usage:
  opi-sztp-agent systemd [flags]

Flags:
  -h, --help             help for systemd
  -o, --options string   sztp-agent args/flags to add into the unit file
  -p, --path string      Path for unit file to be created (default "/etc/systemd/system")

2024/06/24 16:06:34 [ERROR] creating unit file /etc/systemd/system/sztp-agent.service: open /etc/systemd/system/sztp-agent.service: permission denied
```
And it prints the error twice, same with the other commands created. In my personal experience this should only happen if the user specifically specifies ``--help | -h`` with a command and/or subcommand. Not upon command failure, otherwise it drowns out the real error message.

I believe the structure of cobra needs some amendments and fixes to streamline it and make it cleaner. E.g.

```go
func main() {
	command := newCommand()
	if err := command.Execute(); err != nil {
		log.Fatalf(color.InRed("[ERROR] ")+"%s", err.Error())
	}
}

func newCommand() *cobra.Command {
	c := &cobra.Command{
		Use:   "opi-sztp-agent",
		Short: "opi-sztp-agent is the agent command line interface to work with the sztp workflow",
		Run: func(cmd *cobra.Command, _ []string) {
			var err error
			if err != nil {
				log.Fatalf(color.InRed("[ERROR] ")+"%s", err.Error())
			}
		},
	}

	c.AddCommand(cmd.NewSystemdCommand())
	c.AddCommand(cmd.NewEnableCommand())
	return c
}
```
This can be done all under `main()`, and my proposed changes to how we add commands.

I'd be happy to implement this and make all the necessary changes. This way, we'd never have to do anything with the main function unless necessary, and adding commands becomes a little bit more straightforward.

### PR DESCRIPTION
**Pull request by Xeckt**
This PR fixes #413 

I know that using `init()` funcs are not the ideal approach in most cases and is often disregarded and recommended against in Go. But, for the Cobra CLI and the structure of the current source I thought it be necessary for specific reasons. Cobra also recommend this way.

However, I did this a little different to try to avoid the problems with `init()` as much as possible.

Comparatively to the original issue #413:
1. Global slice of `*cobra.command` for the `cmd` pkg 
2. Each command file has an `init()` function that appends the command to the global slice, this is separate from the command functions themselves

This way, we kept the original structure of the command functions so that `init()` doesn't interfere with the testing of the commands. And, if need be, the functions can be used elsewhere.

I decided this was the better approach for `init()` to make it only valuable for the command slice, and not replace the functions directly with `init()`.

I have also renamed the functions for simpler naming, rather than `NewXCommand`, we just use the command name directly as the function name. So for the `disable` command it's now `func Disable()`. I decided this was a nicer change as everything inside `cmd` is obvious it's for the CLI and not for anything else. 

There is a new file `cli.go` inside `cmd` which is handling everything primarily cli related. Only `main` is used to call the root command which sets up everything else.

I've removed the `checkErr(err)` from each command as well, as we have `RunE()` functions on every command, so they always return the error and will always return back to the root command caller and print from there. This stops the issue with double printing the error text and help text unnecessarily. If we need formatted results we can just format an error and return it. 

I've ensured all commits are signed off and have the correct naming, linter passed all checks locally as well.

Let me know if you would like anything amended here!

### CODE REVIEW

**Comment by Xeckt** [comment]
I can fix the unused vars, just let me know if you're okay with the changes and I'll squash then do it all in one.

**Comment by Xeckt** [comment]
I can fix the unused vars, just let me know if you're okay with the changes and I'll squash then do it all in one.

**Review by glimchb — COMMENTED** [review]
I definitely like `cobra.CheckErr(err)` removal
I don;t see any benefit yet for the `init` approach... what exactly are we saving here ? 
seems same amount of code... and we only have handful commands, so maintenance should not be a problem.. 

wdyt ?

**Comment by glimchb** [comment]
how about merging https://github.com/opiproject/sztp/pull/424/commits/d409fb5d4266fdc019b901825d37a0b491d54a38 separately and then coming back to this PR and discussing it a little more...

**Comment by Xeckt** [comment]
> I definitely like `cobra.CheckErr(err)` removal I don;t see any benefit yet for the `init` approach... what exactly are we saving here ? seems same amount of code... and we only have handful commands, so maintenance should not be a problem..
> 
> wdyt ?

I agree maintenance right now isn't a problem.

The benefit of init is just to simplify adding commands.

I wanted to set this up ahead of time, so that the process of anyone coming to the source and/or adding new commands is far more straightforward. So think of it as a future proof approach, [and it's a suggested approach by Cobra.](https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#create-additional-commands)

We don't have to keep adding `cmd.Add(Command)` in the root cobra command either when a new command is added, you do it while you make the new command. It just simplifies the approach while maintaining the same functionality. 

While it may not be a huge difference now, with larger sources and CLI, it definitely makes a difference down the line. I've worked on a source that had over 50 commands including subcommands, and we had implemented this approach early and it saved a lot of headache and 50+ lines of `AddCommand` in one place.

I'm more than happy to merge the error fixes for now if you prefer while we discuss.

**Review by glimchb — APPROVED** [review]
@Xeckt you made a good argument, I can merge it, please fix `commitlint`

**Comment by glimchb** [comment]
@Xeckt please rebase and fix `commitlint`

**Comment by Xeckt** [comment]
I will, I am just at work at the moment, so will do this later.

**Comment by glimchb** [comment]
@Xeckt please squash commit and fix `commitlint`

**Comment by Xeckt** [comment]
> @Xeckt please squash commit and fix `commitlint`

I know, already working on it, my local git seems to have gotten all messed up, resolving the issue now.

**Comment by Xeckt** [comment]
Really weird issue. My fork was synced with main and local changes had no conflicts but kept conflicting on git. Had to rebase with the recent pushes to upstream and fix the HEAD then refix the conflicts... Should all be good now.

**Comment by glimchb** [comment]
See `This branch cannot be rebased due to conflicts`
I can't merge it

**Comment by Xeckt** [comment]
Done

**Pull request merged**
